### PR TITLE
add support for google analytics userId

### PR DESCRIPTION
**Pull request by salimhb**
See:
https://developers.google.com/analytics/devguides/collection/analyticsjs
/user-id

### CODE REVIEW

**Comment by jhilden** [comment]
looks good from my side.  :+1: for merging

**Comment by kangguru** [comment]
please no version bumping in PRs

File: lib/rack/tracker/google_analytics/google_analytics.rb [hunk 1]
```diff
@@ -28,6 +28,19 @@ def tracker
...
+      tracker_options = {}
+
+      tracker_options[:cookieDomain] = options[:cookie_domain] if options[:cookie_domain]
+
+      user_id = options[:user_id].call(env) if options[:user_id]
+      tracker_options[:userId] = "#{user_id}" if user_id.present?
+
+      tracker_options
+    end
+  end
```
Comment (kangguru) [inline lib/rack/tracker/google_analytics/google_analytics.rb]: could this maybe more generic? always camelize the keys, support `#call` on every option....

File: spec/handler/google_analytics_spec.rb [hunk 2]
```diff
@@ -93,6 +121,15 @@ def env
     end
   end
 
+  describe "with user_id tracking" do
```
Comment (jhilden) [inline spec/handler/google_analytics_spec.rb]: isn't that something that should only be tested in the integration spec?

Comment (salimhb) [inline spec/handler/google_analytics_spec.rb]: the cookieDomain thingy is tested here too.
both can be tested again in the integration spec too.

Comment (jhilden) [inline spec/handler/google_analytics_spec.rb]: ok i'm just a bit confused as to what i should test where

File: lib/rack/tracker/google_analytics/google_analytics.rb (hunk 1 above)
Comment (salimhb) [inline lib/rack/tracker/google_analytics/google_analytics.rb]: good point and useful for the cookie_domain with an i18n multihost setup 17000d348411e38d989e90e7bc705d22f4b05dad

File: lib/rack/tracker/google_analytics/google_analytics.rb [hunk 3]
```diff
@@ -28,6 +31,18 @@ def tracker
     options[:tracker].respond_to?(:call) ? options[:tracker].call(env) : options[:tracker]
   end
 
+  def tracker_options
+    @tracker_options ||= begin
+      tracker_options = {}
```
Comment (kangguru) [inline lib/rack/tracker/google_analytics/google_analytics.rb]: i know its a matter of taste, but personally i don't like these temporary variables, `tap` is your fried to get rid of 'em

**Comment by kangguru** [comment]
could we agree on _not_ merging stuff that is under discussion and not reviewed completely?

**Comment by jhilden** [comment]
I will be more careful next time.

**Pull request merged**